p2p/raintree/network.go

```diff
@@ -50,39 +52,64 @@ func NewRainTreeNetwork(addr cryptoPocket.Address, addrBook typesPre2P.AddrBook)
 		log.Println("[ERROR] Error initializing rainTreeNetwork: ", err)
 	}
 
-	return typesPre2P.Network(n)
+	return types2.Network(n)
 }
 
 func (n *rainTreeNetwork) NetworkBroadcast(data []byte) error {
 	return n.networkBroadcastAtLevel(data, n.maxNumLevels, getNonce())
 }
 
-func (n *rainTreeNetwork) networkBroadcastAtLevel(data []byte, level uint32, nonce uint64) error {
-	// This is handled either by the cleanup layer or redundancy layer
-	if level == 0 {
-		return nil
-	}
+func (n *rainTreeNetwork) networkBroadcastAtLevel(data []byte, level int32, nonce uint64) error {
+	var addr1, addr2 cryptoPocket.Address
+	var ok bool
 
-	msg := &typesPre2P.RainTreeMessage{
+	msg := &types2.RainTreeMessage{
 		Level: level,
 		Data:  data,
 		Nonce: nonce,
 	}
+	// This is handled either by the redundancy layer
+	if level == 0 {
+		if n.redundancyLayerEnabled {
+			// redundancy layer is simply one final send to the original +1/3 && -1/3
+			level = n.maxNumLevels
+			// ensure not an echo-chamber
+			msg.Level = -1
+		} else {
+			if err := n.demote(msg); err != nil {
```

Comment:
I don't think we need to call `demote`  if level == 0?

Reply:
Cleanup layer is -1